### PR TITLE
Added demo projects for the annotations approach

### PR DESCRIPTION
**Pull request by timschlechter**
I've added two simple demo projects:
- one using the basic annotations package (Nancy.Swagger.Annotations.Demo)
- one using the annotations package with Json.NET (Nancy.Swagger.Annotations.Demo.JsonNet)

They both 'implement' a very small part of the Petstore API, as used in the [Swagger-UI demo](http://petstore.swagger.io/). I can fill in the gaps later, but first I'm having some trouble with the default registrations in Nancy.Swagger.SwaggerRegistrations.cs:

``` csharp
public SwaggerRegistrations()
{
    //RegisterWithDefault<ISwaggerMetadataProvider>(typeof(DefaultSwaggerMetadataProvider));
    RegisterWithDefault<ISwaggerMetadataConverter>(typeof(DefaultSwaggerMetadataConverter));
    RegisterWithDefault<ISwaggerModelCatalog>(typeof(DefaultSwaggerModelCatalog));
    RegisterAll<ISwaggerModelDataProvider>();
}
```

The commented line causes an exception in the Json.NET demo project, because there are two ISwaggerMetadataProvider discovered by `RegisterWithDefault`: `SwaggerAnnotationsProvider` and `JsonNetAnnotationsProvider`.  I'm not sure how to avoid or deal with this. Is this meant behaviour of the RegisterWithDefault functions in Nancy? As i look at Nancy's code I think it is, but how/what should I do to avoid this exception?

The thing I'm basically trying to do is to get the `JsonNetAnnotationsProvider` registered as the `ISwaggerMetadataProvider` to be used in the application. 
`JsonNetAnnotationsProvider` should encapsulate an `ISwaggerMetadataProvider`, and appends extra info to the models provided by the encapsulated ISwaggerMetadataProvider```. 

Could someone help me by explaining me what's the best way to achieve this in Nancy's IoC container?

### CODE REVIEW

**Comment by khellang** [comment]
Yeah. The problem is that the `RegisterWithDefault<T>(Type)` method is for registering a single service for that interface, so it'll throw if it finds multiple implementations. It basically says "register whatever service you can find that implements the `ISwaggerMetadataProvider` interface and fall back to the `DefaultSwaggerMetadataConverter` if you can't find anything else".

Nancy doesn't have a built-in IoC container. You can bring your own (BYOC :wink:). It's just a thin abstraction for simple registration of types that gets passed to the real container though the bootstrapper at startup.

Basically what you're doing is you want to decorate the scanned `ISwaggerMetadataProvider` with a `JsonNetMetadataProvider`? What about just registering multiple `ISwaggerMetadataProvider`s and change the implementation to support `IEnumerable<ISwaggerMetadataProvider>` instead? Would that work?

**Comment by TimSchlechter** [comment]
Mwah I would go as far as extending Nancy for this, I think I can work around this by explicitly pass an instance of ```ISwaggerMetadataProvider``` I would like to use within the ```JsonNetMetadataProvider```

But if I understand correct an exception is thrown when using RegisterWithDefault<ISomething>``` and an assembly is loaded which contains more than one class implementing ```ISomething```. And there is not really a way to disable the default registration from the outside a the moment.
 
Would it be ok if I removed the default registration for ```ISwaggerMetadataProvider``` in Nancy.Swagger?

**Comment by yahehe** [comment]
Closing as this is now probably very out of date. The demo projects have been updated and wiki documentation has been added for the new version.
